diff --git a/Source/Core/VideoCommon/CMakeLists.txt b/Source/Core/VideoCommon/CMakeLists.txt
index 7976e6974..97b91379a 100644
--- a/Source/Core/VideoCommon/CMakeLists.txt
+++ b/Source/Core/VideoCommon/CMakeLists.txt
@@ -84,9 +84,9 @@ endif()
 if(_M_X86)
 	set(SRCS ${SRCS} x64TextureDecoder.cpp VertexLoaderX64.cpp)
 elseif(_M_ARM_64)
-	set(SRCS ${SRCS} VertexLoaderARM64.cpp TextureDecoder_Generic.cpp)
+	set(SRCS ${SRCS} VertexLoaderARM64.cpp GenericTextureDecoder.cpp)
 else()
-	set(SRCS ${SRCS} TextureDecoder_Generic.cpp)
+	set(SRCS ${SRCS} GenericTextureDecoder.cpp)
 endif()
 
 add_dolphin_library(videocommon "${SRCS}" "${LIBS}")
